dataland-frontend/tests/component/components/general/SlideShow.cy.ts

```diff
@@ -0,0 +1,86 @@
+import TestSlideShow from "./TestSlideShow.vue";
+
+const SLIDE_DELTA = 440;
+
+describe("Component test for the SlideShow component", () => {
+  it("Check if the buttons are working", () => {
+    cy.mountWithPlugins(TestSlideShow, {}).then(() => {
+      assertSlidesPosition(slidesSelector);
+      cy.get(rightButtonSelector).click();
+      assertSlidesPosition(slidesSelector, 1);
+      cy.get(rightButtonSelector).click();
+      assertSlidesPosition(slidesSelector, 2);
+      cy.get(rightButtonSelector).click();
+      assertSlidesPosition(slidesSelector, 2);
+      cy.get(leftButtonSelector).click();
+      assertSlidesPosition(slidesSelector, 1);
+      cy.get(leftButtonSelector).click();
+      assertSlidesPosition(slidesSelector, 0);
+      cy.get(leftButtonSelector).click();
+      assertSlidesPosition(slidesSelector, 0);
+      cy.get(rightButtonSelector).click();
+      assertSlidesPosition(slidesSelector, 1);
+    });
+  });
+
+  it("Check if dragging is working", () => {
+    cy.mountWithPlugins(TestSlideShow, {}).then(() => {
+      assertSlidesPosition(slidesSelector);
+      dragSlideTo(genericSlideSelector, 1, leftOffset);
+      assertSlidesPosition(slidesSelector, 1);
+      dragSlideTo(genericSlideSelector, 2, leftOffset);
+      assertSlidesPosition(slidesSelector, 2);
+      dragSlideTo(genericSlideSelector, 2, leftOffset + SLIDE_DELTA);
+      assertSlidesPosition(slidesSelector, 1);
+      dragSlideTo(genericSlideSelector, 1, leftOffset + SLIDE_DELTA);
+      assertSlidesPosition(slidesSelector, 0);
+    });
+  });
+
+  it("Check if dragging is disabled on large screens", () => {
+    cy.mountWithPlugins(TestSlideShow, {}).then(() => {
+      cy.viewport(1900, 800);
+      assertSlidesPosition(slidesSelector);
+      dragSlideTo(genericSlideSelector, 1, leftOffset);
+      assertSlidesPosition(slidesSelector);
+    });
+  });
+});
+
+const slidesSelector = ".test__slides";
+const leftButtonSelector = "button[aria-label='Previous slide']";
+const rightButtonSelector = "button[aria-label='Next slide']";
+const genericSlideSelector = ".test__slide";
+const leftOffset = 10;
+const topOffset = 10;
+
+/**
+ * Checks that a slide show is centered at a given slide
+ * @param slidesSelector the selector for the slides wrapper
+ * @param centerSlide the index of the slide that is expected to be in the center
+ * @param initialCenterSlide the index of the slide that was centered initially
+ */
+function assertSlidesPosition(slidesSelector: string, centerSlide?: number, initialCenterSlide = 0): void {
```

Comment:
Why is the `slidesSelector` a param, if it is always the same?

Reply:
Also `initialCenterSlide` is always 0 across the whole test. This makes sense of course, because it is a prop of the mounted SlideShow.

But why is it a param in this function?

Reply:
A releict as the one explained below.

Reply:
Removed